libsolidity/formal/ModelChecker.cpp

```diff
@@ -146,3 +155,43 @@ solidity::smtutil::SMTSolverChoice ModelChecker::availableSolvers()
 #endif
 	return available;
 }
+
+SMTSolverChoice ModelChecker::checkRequestedSolvers(SMTSolverChoice _enabled, ErrorReporter& _errorReporter)
+{
+	SMTSolverChoice availableSolvers{ModelChecker::availableSolvers()};
+
+	if (_enabled.cvc4 && !availableSolvers.cvc4)
+	{
+		_enabled.cvc4 = false;
+		_errorReporter.warning(
+			4902_error,
+			SourceLocation(),
+			"Solver CVC4 was selected for SMTChecker but it is not available."
+		);
+	}
+
+	if (_enabled.eld && !availableSolvers.eld)
+	{
+		_enabled.eld = false;
+		_errorReporter.warning(
+			4458_error,
+			SourceLocation(),
+			"Solver Eldarica was selected for SMTChecker but it was not found in the system."
```

Comment:
Like I said in https://github.com/ethereum/solidity/pull/12969#discussion_r1029406607, can we give a different message here when the `ifdef` is false?